read-fonts/src/tables/variations.rs

```diff
@@ -850,11 +884,180 @@ where
+        }
     }
 }
 
+impl TupleVariation<'_, GlyphDelta> {
+    pub fn accumulate_dense_deltas<D: PointCoord>(
+        &self,
+        deltas: &mut [Point<D>],
+        scalar: Fixed,
+    ) -> Result<(), ReadError> {
+        let (_, packed_deltas) = self.point_numbers_and_packed_deltas();
+        let mut cursor = packed_deltas.cursor();
+        if scalar == Fixed::ONE {
+            // scalar of 1.0 is common so avoid the costly conversions and
+            // multiplications per coord
+            read_dense_deltas(&mut cursor, deltas, |delta, new_delta| {
+                delta.x += D::from_i32(new_delta);
+            })?;
+            read_dense_deltas(&mut cursor, deltas, |delta, new_delta| {
+                delta.y += D::from_i32(new_delta);
+            })?;
+        } else {
+            read_dense_deltas(&mut cursor, deltas, |delta, new_delta| {
+                delta.x += D::from_fixed(Fixed::from_i32(new_delta) * scalar);
+            })?;
+            read_dense_deltas(&mut cursor, deltas, |delta, new_delta| {
+                delta.y += D::from_fixed(Fixed::from_i32(new_delta) * scalar);
+            })?;
+        }
+        Ok(())
+    }
+
+    pub fn accumulate_sparse_deltas<D: PointCoord>(
+        &self,
+        deltas: &mut [Point<D>],
+        flags: &mut [PointFlags],
+        scalar: Fixed,
+    ) -> Result<(), ReadError> {
+        let (point_numbers, packed_deltas) = self.point_numbers_and_packed_deltas();
+        let mut cursor = packed_deltas.cursor();
+        let count = point_numbers.count() as usize;
+        if scalar == Fixed::ONE {
+            // scalar of 1.0 is common so avoid the costly conversions and
+            // multiplications per coord
+            read_sparse_deltas(&mut cursor, &point_numbers, count, |ix, new_delta| {
+                if let Some((delta, flag)) = deltas.get_mut(ix).zip(flags.get_mut(ix)) {
+                    delta.x += D::from_i32(new_delta);
+                    flag.set_marker(PointMarker::HAS_DELTA);
+                }
+            })?;
+            read_sparse_deltas(&mut cursor, &point_numbers, count, |ix, new_delta| {
+                if let Some(delta) = deltas.get_mut(ix) {
+                    delta.y += D::from_i32(new_delta);
+                }
+            })?;
+        } else {
+            read_sparse_deltas(&mut cursor, &point_numbers, count, |ix, new_delta| {
+                if let Some((delta, flag)) = deltas.get_mut(ix).zip(flags.get_mut(ix)) {
+                    delta.x += D::from_fixed(Fixed::from_i32(new_delta) * scalar);
+                    flag.set_marker(PointMarker::HAS_DELTA);
+                }
+            })?;
+            read_sparse_deltas(&mut cursor, &point_numbers, count, |ix, new_delta| {
+                if let Some(delta) = deltas.get_mut(ix) {
+                    delta.y += D::from_fixed(Fixed::from_i32(new_delta) * scalar);
+                }
+            })?;
+        }
+        Ok(())
+    }
+}
+
+fn read_dense_deltas<T>(
```

Comment:
ditto, given that we're adding this fn specifically because to address performance concerns (and thus it was not the most obvious implementation approach) I think it's worth noting that, and explaining briefly why it's better.